src/main/scala/org/broadinstitute/dsde/workbench/avram/db/DbReference.scala

```diff
@@ -0,0 +1,54 @@
+package org.broadinstitute.dsde.workbench.avram.db
+
+import org.apache.commons.dbcp2.BasicDataSource
+import org.broadinstitute.dsde.workbench.avram.config.DbcpDataSourceConfig
+import slick.dbio.DBIO
+import slick.jdbc.{JdbcProfile, TransactionIsolation}
+
+import scala.concurrent.Future
+import AvramPostgresProfile.api._
+
+case class DbReference(private val dataSourceConfig: DbcpDataSourceConfig) {
+
+  val dataAccess = new DataAccess(AvramPostgresProfile)
+
+  def makeDbcpDataSource: BasicDataSource = {
+    // See https://commons.apache.org/proper/commons-dbcp/configuration.html for configuration options and defaults
+    val ds = new BasicDataSource()
+    ds.setDriverClassName(dataSourceConfig.driverClassName)
+    ds.setUrl(dataSourceConfig.url)
+    ds.setUsername(dataSourceConfig.username)
+    ds.setPassword(dataSourceConfig.password)
+    ds.setMaxTotal(dataSourceConfig.maxTotal)
+    ds
+  }
+  val dbcpDataSource: BasicDataSource = makeDbcpDataSource
+
+  val database = Database.forDataSource(dbcpDataSource, Option(dbcpDataSource.getMaxTotal), AsyncExecutor("Avram Executor", dataSourceConfig.slickNumThreads, dataSourceConfig.slickQueueSize))
+
+  def inTransaction[T](f: (DataAccess) => DBIO[T], isolationLevel: TransactionIsolation = TransactionIsolation.ReadCommitted): Future[T] = {
+    database.run(f(dataAccess).transactionally.withTransactionIsolation(isolationLevel))
```

Comment:
I'd rather we leave out setting the transaction isolation level, even if we're defaulting to the database default, until we find we need to change it (if ever).